internal/project/session.go

```diff
@@ -91,14 +103,6 @@ type Session struct {
 	pendingATAChanges   map[tspath.Path]*ATAStateChange
 	pendingATAChangesMu sync.Mutex
 
-	// snapshotUpdateCancel is the cancelation function for a scheduled
-	// snapshot update. Snapshot updates are debounced after file watch
-	// changes since many watch events can occur in quick succession
-	// during `npm install` or git operations.
-	// !!! This can probably be replaced by ScheduleDiagnosticsRefresh()
-	snapshotUpdateCancel context.CancelFunc
-	snapshotUpdateMu     sync.Mutex
```

Comment:
Removing this and the related code was a bit of a drive-by cleanup. There’s really no reason to trigger a snapshot change internally; the only time we were using this we _also_ scheduled a diagnostics refresh, and the subsequent diagnostics request would have updated the snapshot as needed. It's simpler and cleaner to let snapshot updates be purely reactive to a client request/notification.